pkg/dynamic/client.go

```diff
@@ -87,3 +95,16 @@ func (c *knDynamicClient) ListSourcesTypes() (*unstructured.UnstructuredList, er
 	options.LabelSelector = sourcesLabels.String()
 	return c.ListCRDs(options)
 }
+
+func (c knDynamicClient) RawClient() dynamic.Interface {
+	return c.client
+}
+
+// CreateFakeKnDynamicClient gives you a dynamic client for testing contianing the given objects.
+func CreateFakeKnDynamicClient(testNamespace string, objects ...runtime.Object) KnDynamicClient {
```

Comment:
I have put the Fake helper method in a different package, even, because you're right it's nice to have test code decoupled from mainline code. On the other hand, I really like the mocking scheme here. Using it was refreshing and much nicer — I would much rather specify the objects in my "fake server" than specify _every_ call to it.. I'll keep this mocking scheme. Might even look to use it in more places.